### PR TITLE
Add sentry pipeline tracing

### PR DESCRIPTION
**Pull request by jarrel-b**
This PR adds performance tracing to the pipeline: [example trace](https://usegallery.sentry.io/performance/tokenprocessing:a5bfc7adf1a74c4cb73fe95f99fb5a8a/?environment=local&project=6771759&query=http.method%3APOST&referrer=performance-transaction-summary&showTransactions=recent&statsPeriod=24h&transaction=POST+%2Fmedia%2Fprocess%2Ftoken&unselectedSeries=p100%28%29)

### CODE REVIEW

File: tokenprocessing/handler.go [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
 )
 
 func handlersInitServer(router *gin.Engine, tp *tokenProcessor, mc *multichain.Provider, repos *postgres.Repositories, throttler *throttle.Locker) *gin.Engine {
 	mediaGroup := router.Group("/media")
-	mediaGroup.POST("/process", processMediaForUsersTokens(tp, repos.TokenRepository, repos.ContractRepository, throttler))
+	mediaGroup.POST("/process", func(c *gin.Context) {
+		if hub := sentryutil.SentryHubFromContext(c); hub != nil {
+			hub.Scope().AddEventProcessor(sentryutil.SpanFilterEventProcessor(c, 1000, 1*time.Millisecond, 8, true))
+		}
+		processMediaForUsersTokens(tp, repos.TokenRepository, repos.ContractRepository, throttler)
```
Comment (benny-conn) [inline tokenprocessing/handler.go:19]: Does this work as expected? `processMediaForUsersTokens` returns a func that takes in a gin.Context so maybe we have to do `processMediaForUsersTokens(tp, repos.TokenRepository, repos.ContractRepository, throttler)(c)` ?

File: tokenprocessing/media.go [hunk 2]
```diff
@@ -202,7 +202,8 @@ func cacheObjectsForMetadata(pCtx context.Context, metadata persist.TokenMetadat
 
 	// neither download worked, unexpectedly
 	if (animCh == nil || (animResult.err != nil && len(animResult.cachedObjects) == 0)) && (imgCh == nil || (imgResult.err != nil && len(imgResult.cachedObjects) == 0)) {
-		defer persist.TrackStepStatus(pCtx, &pMeta.NothingCachedWithErrors, "NothingCachedWithErrors")()
+		persist.TrackStepStatus(pCtx, &pMeta.NothingCachedWithErrors, "NothingCachedWithErrors")
```
Comment (benny-conn) [inline tokenprocessing/media.go:205]: Is there a reason why this one and the next one don't do `traceCallback, ctx :=` and `defer traceCallback()`?

**Review by benny-conn — APPROVED** [review]
Looks great, just a few small comments!

File: tokenprocessing/handler.go (hunk 1 above)
Comment (jarrel-b) [reply tokenprocessing/handler.go:19]: Ah good catch! You're right. I originally had this factored out but decided to do it inline

File: tokenprocessing/media.go (hunk 2 above)
Comment (jarrel-b) [reply tokenprocessing/media.go]: Whoops, you're right this should call `traceCallback()`.

**Pull request merged**